coffee-chats/src/main/webapp/src/util/fetch.ts

```diff
@@ -0,0 +1,25 @@
+import React from "react";
+
+/**
+ * A hook that fetches JSON data from a URL using a GET request
+ *
+ * @param url: URL to fetch data from
+ */
+export function useFetch(url: string): any {
+  const [data, setData] = React.useState(null);
```

Comment:
Using an internal persistent state doesn't seem like a great idea here, especially when things are running asynchronously I would worry about two calls corrupting each other before returning. Maybe best to just return a Promise?

Reply:
Do promises play nice with react components? I was looking to create a hook just for that reason. Anyways, it should only be able to race with itself within the same component (and I fixed that already).

Reply:
In short promises do play nice with components, but I see that this is for consistency now. LGTM